WebApp/handler/AuthenticationHandler.py

```diff
@@ -1,79 +1,132 @@
+from DIRAC.ConfigurationSystem.Client.Helpers.Resources import getIdPOption
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import getUsernameForID
+
+try:
+  from OAuthDIRAC.FrameworkSystem.Client.OAuthManagerClient import OAuthManagerClient
+  oauth = OAuthManagerClient()
+except ImportError:
+  oauth = None
+
 
 class AuthenticationHandler(WebHandler):
 
   AUTH_PROPS = "all"
 
-  # Send mail to administrators
   @asyncGen
   def web_sendRequest(self):
+    """ Send mail to administrators
+    """
     typeAuth = str(self.request.arguments["typeauth"][0])
     loadValue = self.request.arguments["value"]
     addresses = Conf.getCSValue('AdminsEmails')
-    NotificationClient().sendMail(
-        addresses, subject="Request from %s %s" %
-        (loadValue[0], loadValue[1]), body='Type auth: %s, details: %s' %
-        (typeAuth, loadValue))
+    subject = "Request from %s %s" % (loadValue[0], loadValue[1])
+    body = 'Type auth: %s, details: %s' % (typeAuth, loadValue)
+    result = NotificationClient().sendMail(addresses, subject=subject, body=body)
+    self.finish(result)
+
+  @asyncGen
+  def web_getAuthNames(self):
+    """ Get list of enable authentication types
+    """
+    self.finish(Conf.getAuthNames())
 
-  # Get information from web.cfg about auth types
   @asyncGen
   def web_getAuthCFG(self):
+    """ Get option from IdP
+    """
     typeAuth = str(self.request.arguments["typeauth"][0])
     loadValue = self.request.arguments["value"][0]
-    res = {}
-    if Conf.getCSSections("TypeAuths")['OK']:
-      if typeAuth:
-        if loadValue:
-          if loadValue == 'all':
-            res = Conf.getCSOptionsDict("TypeAuths/%s" % typeAuth).get('Value')
-          else:
-            res = Conf.getCSValue("TypeAuths/%s/%s" % (typeAuth, loadValue), None)
-        else:
-          res = Conf.getCSOptions("TypeAuths/%s" % typeAuth)
-      else:
-        res = Conf.getCSSections("TypeAuths")
-    self.write(res)
+    res = getIdPOption(typeAuth, loadValue)
+    if not res:
+      providerType = getIdPOption(typeAuth, 'Type')
+      if providerType == 'OAuth2':
+        if loadValue == 'authorization_url':
+          res = oauth.createAuthRequestURL(typeAuth)
+          if not res['OK']:
+            self.finish(res)
+          res = res['Value']
+    self.finish(S_OK(res))
 
-  # Get current auth type
   @asyncGen
   def web_getCurrentAuth(self):
+    """ Get current authentication type
+    """
     if self.get_secure_cookie("TypeAuth"):
       current = self.get_secure_cookie("TypeAuth")
     else:
       current = 'default'
-    self.write(current)
+    self.finish(current)
+
+  @asyncGen
+  def web_waitOAuthStatus(self):
+    """ Listen authentication status on OAuthDB
+    """
+    state = str(self.request.arguments["state"][0])
+    typeAuth = str(self.request.arguments["typeauth"][0])
+    gLogger.debug('Read authentication status of "%s" session' % state)
```

Comment:
glogger.debug("Read authentication status of ", "%s session" % state)

Reply:
fixed